control/beaconing/happy/sender.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2025 SCION Association, Anapaya Systems
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package happy
+
+import (
+	"context"
+	"net"
+
+	"github.com/scionproto/scion/control/beaconing"
+	"github.com/scionproto/scion/pkg/addr"
+	"github.com/scionproto/scion/pkg/connect/happy"
+	"github.com/scionproto/scion/pkg/private/serrors"
+	seg "github.com/scionproto/scion/pkg/segment"
+)
+
+// BeaconSenderFactory can be used to create beacon senders.
+type BeaconSenderFactory struct {
+	Connect beaconing.SenderFactory
+	Grpc    beaconing.SenderFactory
+}
+
+// NewSender returns a beacon sender that can be used to send beacons to a remote CS.
+func (f *BeaconSenderFactory) NewSender(
+	ctx context.Context,
+	dstIA addr.IA,
+	egIfID uint16,
+	nextHop *net.UDPAddr,
+) (beaconing.Sender, error) {
+	connectSender, err := f.Connect.NewSender(ctx, dstIA, egIfID, nextHop)
+	if err != nil {
+		return nil, err
```

Comment:
Does it make sense to wrap those errors to make it easier to understand which of the two (connect sender or grpc sender) failed?